### DISCUSSION TITLE
Why are there so many different versions of apps in the main homebrew-cask?

### DISCUSSION THREAD

**Discussion by wytsai7660** [Homebrew/discussions#5330] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: c2ed3327c605c3e738359c9807b8f4cd6fec09eb
Last commit: 2 days ago
Core tap JSON: 01 May 02:05 UTC
Core cask tap JSON: 01 May 02:05 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BAT_THEME: Visual Studio Dark+
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.25pP21OnX8/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.45.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

Recently, when I run `brew update`, there are a lot of new different versions of apps showing up (beta, nightly,...). I can't really see the purpose of doing so. Don't we already have [homebrew-cask-versions](https://github.com/Homebrew/homebrew-cask-versions)?

**Comment by p-linnane** [comment, marked as answer]
We are migrating formulae to from `homebrew-cask-versions` to the main repo. We will be deprecating the versions repo once this is complete.

**Reply by ngocphamm** [reply]
@p-linnane Is there a command to show which casks I have installed have been migrated, so I can reinstall them with the new "names"?

**Reply by SMillerDev** [reply]
Brew will handle that already under the hood.
